src/metabase/api/database.clj

```diff
@@ -1030,9 +1028,10 @@
                     e))]
       (throw (ex-info (ex-message ex) {:status-code 422}))
       (do
-        (future
-          (sync-metadata/sync-db-metadata! db)
-          (analyze/analyze-db! db))
+        (sync/submit-task!
+         (fn []
+           (sync/sync-db-metadata! db)
+           (sync/analyze-db! db)))
```

Comment:
I fixed what was almost certainly a bug, sync has its own threadpool and almost everybody is respectful and uses it, but this code did not.